### PR TITLE
Awesome books Module

### PR DESCRIPTION
**Pull request by karam084**
* Project requirements

- Create a new directory and initialize it with git.
- Use npm init -y command to create package.json file.
- Create the entry point for your JavaScript code called index.js
- Create index.css file for your styles.
- Create the main index.html file. Link your index.js and index.css files there.
- Create a directory called modules.
- Use ES6 syntax.
- Divide your code into modules. Save each module in a separate file in the modules dir. Import modules in the index.js file. For this exercise - the more modules the better!
- Use let and const in a correct way.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import data from './data.js';
-
+import data from './modules/book.js';
+import { DateTime } from './node_modules/luxon/build/es6/luxon.js';
```
Comment (Gambit142) [inline index.js]: - [x] This line of code is stopping the synchronous flow of your logic in the `index.js` file. Kindly remove it. The project requirement states that you should use the Luxon package as a `javascript` file and then import the `DateTime` object. 

_**Hint:**_ When you add `Luxon` as a minified or full `.js` file, you can import it using:

`import { DateTime } from './luxonfilename'`

File: index.html [hunk 2]
```diff
@@ -23,7 +23,7 @@
           </li>
         </ul>
       </nav>
-      <div id="date">5pm eastern time</div>
+      <div id="time">Febuary 9th 2022, 1:43pm</div>
```
Comment (Gambit142) [inline index.html]: - [x] Kindly remove this div. The current date and time under the navbar should be provided by Luxon library and not hardcoded. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @karam084 👋, 

Good job so far!

### Highlight
- Linters checks passed ✔️ 
- Professional README.md file ✔️ 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- Kindly separate the code in the `index.js` into modules and then import them and use them in the `index.js` file. You can have one module responsible for handling the addition of books, another for removing and another for local storage. It is left for you to decide how you want to separate them.
- The code for adding and removing a book is not functioning properly. Kindly fix it

- _Kindly check the comments under the review for other required changes._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: index.html [hunk 3]
```diff
@@ -4,7 +4,7 @@
     <meta charset="UTF-8" />
     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <link rel="stylesheet" href="./main.css" />
+    <link rel="stylesheet" href="./index.css" />
```
Comment (BenMukebo) [inline index.html:7]: - [x] Greate job on implementing changes suggested by the previous reviewer, however, your app is still not working yet, this is what actually displaying on your website, `list, add new and contact` nav links are not working, to mean I can not run you project, please fix that and resubmit the work.
yuu

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi **@karam084,**

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight 💪🏻

- No linter errors. ✔️ 
- Good commit messages ✔️ 
- Professional README ✔️ 
- Descriptive README file.  ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏 💻  

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html (hunk 3 above)
Comment (karam084) [reply index.html:7]: Good day, 
I think you need to install node_modules on your local machine to run luxon.js so I add node modules in .ignore file.
Thank you

**Review by Davidosky007 — COMMENTED** [review]
Hello @karam084  👋,

_This is an additional review as a result of the second opinion request._

You have done a great job implementing almost all the requirements for this project 👍 

- for how to properly use and add the luxon library i strongly suggest you follow the requirement on this [requirement link here ](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md#:~:text=Display%20the%20current,your%20future%20job.) adding the library this way makes it very easy to use and review instead of fetching the library script from ` /node_modules/luxon/src/luxon.js` you can download the script ( luxon.js ) [From this link](https://moment.github.io/luxon/es6/luxon.js) save it in your project then you import it as a module ( ie `import { DateTime } from "luxon";`) 
- [OPTIONAL] You could have added step by step instructions in your readme file on how to run the project successfully in your case you could have added ` npm install` so that visitors know that they have to install all dependencies needed for the app to run properly.
## Note: Above are solutions you could take to resolve this issue but i strongly suggest you implement the first as it is the required way to add the luxon library 🙏 
Kindly fix this issue and submit it for review 👍 
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away :)

Keep rocking 🚀!

**Review by dasileker — APPROVED** [review]
### **Hi @karam084,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**